hazelcast/src/main/java/com/hazelcast/monitor/LocalIndexStats.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.monitor;
+
+/**
+ * Provides local statistics for an index to be used by {@link MemberState}
+ * implementations.
+ */
+public interface LocalIndexStats extends LocalInstanceStats {
+
+    /**
+     * Returns the total number of queries served by the index.
+     * <p>
+     * To calculate the index hit rate just divide the returned value by a value
+     * returned by {@link LocalMapStats#getQueryCount()}.
+     * <p>
+     * The returned value may be less than the one returned by {@link
+     * #getHitCount()} since a single query may hit the same index more than once.
+     */
+    long getQueryCount();
+
+    /**
+     * Returns the total number of hits into the index.
+     * <p>
+     * The returned value may be greater than the one returned by {@link
+     * #getQueryCount} since a single query may hit the same index more than once.
+     */
+    long getHitCount();
+
+    /**
+     * Returns the average hit latency for the index.
+     */
+    long getAverageHitLatency();
```

Comment:
what about having the related units in the name of the methods like `getAverageHitLatencyNanos`, `getMemoryCostInBytes `

Reply:
I like the idea, but it's not in line with other stats, see map and cache stats for instance. The javadocs definitely must be improved. WDYT?

Reply:
Up to you but i'm in favor of using units in the names.

Reply:
It would be great if method names had name with units but I believe that more important is to have this stats names consistent with other stats as @taburet said.

Reply:
javadoc updated